### PR TITLE
feat(gatsby-remark-prismjs): Allow global line number config

### PR DESCRIPTION
**Pull request by timbroder**
Hi!

First-time contributor, please let me know if I can improve anything!

Line number support for Prism was added in #5821! 🎆 But, it looks like [showLineNumbers](https://github.com/gatsbyjs/gatsby/pull/5821/files#diff-8a21a1924b6e656de27f987de9deeacdR47) in the main config isn't used so you can [conditionally](https://github.com/gatsbyjs/gatsby/pull/5821#issuecomment-396573481) turn on for specific code blocks when you want it.

This PR turns that high level config back on if you want it. I'm migrating a few hundred posts over from Jekyll and don't want to go through each post turning line numbers on 😸 

CC: @phacks (original line number PR)

### CODE REVIEW

File: packages/gatsby-remark-prismjs/README.md [hunk 1]
```diff
@@ -39,11 +39,13 @@ plugins: [
             // the language "sh" which will highlight using the
             // bash highlighter.
             aliases: {},
-            // This toggles the display of line numbers alongside the code.
+            // This toggles the display of line numbers globally alongside the code.
             // To use it, add the following line in src/layouts/index.js
             // right after importing the prism color scheme:
             //  `require("prismjs/plugins/line-numbers/prism-line-numbers.css");`
             // Defaults to false.
+            // If you with to only show line numbers on certain code blocks,
```
Comment (jgierer12) [inline packages/gatsby-remark-prismjs/README.md]:
```suggestion
            // If you wish to only show line numbers on certain code blocks,
```

**Comment by timbroder** [comment]
@jgierer12 thanks! (oh man the new suggestion features are great)

**Comment by phacks** [comment]
Looks good to me!

**Comment by timbroder** [comment]
Sorry for the close / reopen. Was playing with Githawk and thought it meant close the notification. 

@jgierer12 I'm not sure what the bootstrap failure is, could you advise? Thanks

**Comment by jgierer12** [comment]
I don't think the failing CI has anything to do with this PR, no need to worry about it.

**Comment by phacks** [comment]
@timbroder Could you try to do the following to see if tests pass now?

- do a `git commit --amend` without changing anything on your branch and `push --force` it again?
- if it did not work, maybe `git co master && git pull && git co - && git rebase master`, fixing conflicts if any, and `git push --force` again?

Thanks!

**Comment by timbroder** [comment]
@phacks that did it! Thanks!

**Comment by timbroder** [comment]
@jgierer12 what is the next step to merge this? Thanks

**Comment by jgierer12** [comment]
Sorry, seems like this flew a bit under my radar :sweat_smile: 

Thanks a lot @timbroder :raised_hands:

**Comment by pieh** [comment]
Before publishing this, I have just one potential correction - `showLineNumbersGlobal` makes perfect sense in context of the function, but it sounds a little weird when defining plugin options (where every option there is really "global") and also this changes "API" of plugin a bit (so would need potentially major version bump).

My preferred solution for this would be to keep previous name of the option and alias it in the function:
```diff
module.exports = (
  { markdownAST },
  {
    classPrefix = `language-`,
    inlineCodeMarker = null,
    aliases = {},
    noInlineHighlight = false,
-   showLineNumbersGlobal = false,
+   showLineNumbers: showLineNumbersGlobal = false,
  } = {}
) => {
```
and adjust readme to use `showLineNumbers`. What do you think?

And sorry @timbroder for slow PR handling, I've missed this one too.

**Pull request merged**